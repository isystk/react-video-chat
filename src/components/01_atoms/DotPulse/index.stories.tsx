import { Meta, Story } from '@storybook/react'
import React from 'react'
import DotPulse from './index'

export default {
  title: '01_atoms/DotPulse',
  component: DotPulse,
} as Meta

const Template: Story = (props) => {
  return (
    <DotPulse>
      <div
        style={{
          width: '100%',
          height: '600px',
          background: '#eee',
        }}
      ></div>
    </DotPulse>
  )
}

export const Primary = Template.bind({})
Primary.storyName = 'プライマリ'
Primary.args = {}
