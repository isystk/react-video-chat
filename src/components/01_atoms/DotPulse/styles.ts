import { makeStyles } from '@material-ui/core/styles'

const useStyles = makeStyles((theme) => ({
  dotPulse: {
    position: 'relative',
    left: '-9999px',
    textAlign: 'initial',
    width: '10px',
    height: '10px',
    borderRadius: '5px',
    backgroundColor: '#9880ff',
    color: '#9880ff',
    boxShadow: '9999px 0 0 -5px #9880ff',
    animation: '$dotPulse 1.5s infinite linear',
    animationDelay: '.25s',
    '&::before, &::after': {
      content: '""',
      display: 'inline-block',
      position: 'absolute',
      top: '0',
      width: '10px',
      height: '10px',
      borderRadius: '5px',
      backgroundColor: '#9880ff',
      color: '#9880ff',
    },
    '&::before': {
      boxShadow: '9984px 0 0 -5px #9880ff',
      animation: '$dotPulseBefore 1.5s infinite linear',
      animationDelay: '0s',
    },
    '&::after': {
      boxShadow: '10014px 0 0 -5px #9880ff',
      animation: '$dotPulseAfter 1.5s infinite linear',
      animationDelay: '.5s',
    },
  },
  '@keyframes dotPulseBefore': {
    '0%': {
      boxShadow: '9984px 0 0 -5px #9880ff',
    },
    '30%': {
      boxShadow: '9984px 0 0 2px #9880ff',
    },
    '60%, 100%': {
      boxShadow: '9984px 0 0 -5px #9880ff',
    },
  },
  '@keyframes dotPulse': {
    '0%': {
      boxShadow: '9999px 0 0 -5px #9880ff',
    },
    '30%': {
      boxShadow: '9999px 0 0 2px #9880ff',
    },
    '60%,100%': {
      boxShadow: '9999px 0 0 -5px #9880ff',
    },
  },
  '@keyframes dotPulseAfter': {
    '0%': {
      boxShadow: '10014px 0 0 -5px #9880ff',
    },
    '30%': {
      boxShadow: '10014px 0 0 2px #9880ff',
    },
    '60%,100%': {
      boxShadow: '10014px 0 0 -5px #9880ff',
    },
  },
}))

export { useStyles }
