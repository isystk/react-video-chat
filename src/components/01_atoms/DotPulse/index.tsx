import React, { FC } from 'react'
import { ContainerProps, WithChildren } from 'types'
import { connect } from '@/components/hoc'
import { useStyles } from './styles'

/** DotPulseProps Props */
export type DotPulseProps = WithChildren
/** Presenter Props */
export type PresenterProps = DotPulseProps & { classes }

/** Presenter Component */
const DotPulsePresenter: FC<PresenterProps> = ({
  children,
  classes,
  ...props
}) => (
  <>
    <div className={classes.dotPulse}></div>
  </>
)

/** Container Component */
const DotPulseContainer: React.FC<
  ContainerProps<DotPulseProps, PresenterProps>
> = ({ presenter, children, ...props }) => {
  const classes = useStyles()
  return presenter({
    children,
    classes,
    ...props,
  })
}

export default connect<DotPulseProps, PresenterProps>(
  'DotPulse',
  DotPulsePresenter,
  DotPulseContainer
)
