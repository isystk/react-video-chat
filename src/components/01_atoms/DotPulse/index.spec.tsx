import React from 'react'
import renderer from 'react-test-renderer'
import DotPulse from './index'
import '@testing-library/jest-dom/extend-expect'

describe('DotPulse', () => {
  it('Match Snapshot', () => {
    const component = renderer.create(<DotPulse />)
    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })
})
