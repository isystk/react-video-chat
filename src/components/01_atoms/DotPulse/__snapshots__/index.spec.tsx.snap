// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DotPulse Match Snapshot 1`] = `
<div
  className="makeStyles-dotPulse-1"
/>
`;
